products/revenue_analytics/backend/hogql_queries/test/test_revenue_example_data_warehouse_tables_query_runner.py

```diff
@@ -0,0 +1,107 @@
+            team=self.team,
+            source_id="source_id",
+            connection_id="connection_id",
+            status=ExternalDataSource.Status.COMPLETED,
+            source_type=ExternalDataSource.Type.STRIPE,
+        )
+
+        self.credential = DataWarehouseCredential.objects.create(
+            team=self.team,
+            access_key="test",
+            access_secret="test",
+        )
+
+        self.table = DataWarehouseTable.objects.create(
+            name="stripe_charge",
+            format=DataWarehouseTable.TableFormat.Parquet,
+            team=self.team,
+            credential=self.credential,
+            url_pattern="test://localhost",
+            columns={
+                "id": {"hogql": "StringDatabaseField", "clickhouse": "String", "schema_valid": True},
+                "revenue": {"hogql": "FloatDatabaseField", "clickhouse": "Float64", "schema_valid": True},
+                "timestamp": {"hogql": "DateTimeDatabaseField", "clickhouse": "DateTime", "schema_valid": True},
+            },
+        )
+
+        self.schema = ExternalDataSchema.objects.create(
+            team=self.team,
+            name=STRIPE_DATA_WAREHOUSE_CHARGE_IDENTIFIER,
+            source=self.source,
+            table=self.table,
+            should_sync=True,
+            last_synced_at="2024-01-01",
+        )
+
+    def _run_revenue_example_external_tables_query(self):
+        with freeze_time(self.QUERY_TIMESTAMP):
+            query = RevenueExampleDataWarehouseTablesQuery(revenueTrackingConfig=REVENUE_TRACKING_CONFIG)
+            runner = RevenueExampleDataWarehouseTablesQueryRunner(team=self.team, query=query)
+
+            response = runner.calculate()
+            RevenueExampleDataWarehouseTablesQueryResponse.model_validate(response)
+
+            return response
+
+    def test_no_crash_when_no_data(self):
+        self.table.delete()
+        results = self._run_revenue_example_external_tables_query().results
+
+        assert len(results) == 0
+
+    # TODO: These fail because it'll complain the table doesn't exist
+    # posthog.errors.CHQueryErrorUnknownTable: Code: 60.
+    # DB::Exception: Unknown table expression identifier 'stripe_charge'
+    @patch("posthoganalytics.feature_enabled", return_value=False)
+    def test_database_query(self, feature_enabled_mock):
+        response = self._run_revenue_example_external_tables_query()
+        results = response.results
+
+        assert len(results) == 3
+
+    # TODO: These fail because it'll complain the table doesn't exist
+    # posthog.errors.CHQueryErrorUnknownTable: Code: 60.
+    # DB::Exception: Unknown table expression identifier 'stripe_charge'
+    @patch("posthoganalytics.feature_enabled", return_value=True)
+    def test_database_query_with_currency_conversion(self, feature_enabled_mock):
+        response = self._run_revenue_example_external_tables_query()
+        results = response.results
+
+        assert len(results) == 3
```

Comment:
@PostHog/team-data-warehouse I need some help here. I want to be able to run these locally, but it won't run because:
- Im using a parquet file that doesn't exist, how do I make it exist? Can I make it a CSV instead? I see team experiments put some effort into this, but it's a lot of work, and it requires some Docker internal host handling that I'm not knowledgeable about
- How do I make Clickhouse know about these tables/views? Do I need to create them somehow?

Reply:
I'll skip these for now in the interest of time and will come back in the future. I'll add a `skip` annotation to these tests

Reply:
Im now working on getting these to pass using a local CSV while I don't get an approve from @team-dawa

Reply:
Context here: https://posthog.slack.com/archives/C019RAX2XBN/p1743724035504849